File: sdks/go/cv/inspect.go
package cv

import "fmt"

// Inspect returns the metadata of a .cv file without extracting payloads.
func Inspect(data []byte) (*Metadata, error) {
	ctx, err := loadContext(data)
	if err != nil {
		return nil, err
	}
	xml, err := readMetadataXML(ctx)
	if err != nil {
		return nil, err
	}
	if xml == "" {
		return nil, fmt.Errorf("not a .cv file: no /Metadata stream")
	}
	meta, ok := parseXMP(xml)
	if !ok {
		return nil, fmt.Errorf("not a .cv file: XMP missing required cv: properties")
	}
	return meta, nil
}

File: sdks/python/src/cvfile/inspect.py
"""inspect(): return only the metadata, without extracting payloads."""

from __future__ import annotations

import io

from pypdf import PdfReader
from pypdf.errors import PdfReadError

from cvfile._pdf import read_metadata_xml
from cvfile._types import CvMetadata
from cvfile._xmp import parse_xmp


def inspect(data: bytes) -> CvMetadata:
    try:
        reader = PdfReader(io.BytesIO(data))
    except (PdfReadError, KeyError, ValueError) as err:
        raise ValueError(f"Not a .cv file: failed to parse PDF ({err})") from err
    xml = read_metadata_xml(reader)
    if not xml:
        raise ValueError("Not a .cv file: no /Metadata XMP stream")
    meta = parse_xmp(xml)
    if not meta:
        raise ValueError("Not a .cv file: XMP missing required cv: properties")
    return meta


__all__ = ["inspect"]
